protocol/communities/manager.go

```diff
@@ -3498,66 +3509,19 @@ func (m *Manager) HandleCommunityRequestToJoinResponse(signer *ecdsa.PublicKey,
 		return nil, err
 	}
 
-	communityDescriptionBytes, err := proto.Marshal(request.Community)
-	if err != nil {
-		return nil, err
-	}
-
-	// We need to wrap `request.Community` in an `ApplicationMetadataMessage`
-	// of type `CommunityDescription` because `UpdateCommunityDescription` expects this.
-	//
-	// This is merely for marsheling/unmarsheling, hence we attaching a `Signature`
-	// is not needed.
-	metadataMessage := &protobuf.ApplicationMetadataMessage{
-		Payload: communityDescriptionBytes,
-		Type:    protobuf.ApplicationMetadataMessage_COMMUNITY_DESCRIPTION,
-	}
-
-	appMetadataMsg, err := proto.Marshal(metadataMessage)
-	if err != nil {
-		return nil, err
-	}
-
-	isControlNodeSigner := common.IsPubKeyEqual(community.ControlNode(), signer)
-	if !isControlNodeSigner {
-		m.logger.Debug("signer is not control node", zap.String("signer", common.PubkeyToHex(signer)), zap.String("controlNode", common.PubkeyToHex(community.ControlNode())))
-		return nil, ErrNotAuthorized
-	}
-
-	_, processedDescription, err := m.preprocessDescription(community.ID(), request.Community)
-	if err != nil {
-		return nil, err
-	}
-
-	_, err = community.UpdateCommunityDescription(processedDescription, appMetadataMsg, nil)
-	if err != nil {
-		return nil, err
-	}
-
-	if err = m.handleCommunityTokensMetadata(community); err != nil {
-		return nil, err
-	}
-
```

Comment:
Hmm, don't we want to keep this around for 3 versions, until users upgrade?
(only process it if `CommunityDescriptionMessage` is missing)

Reply:
That logic is the root cause of the problem. If we keep it, syncing/backup will not work as expected. I don't believe removing this part of the code will be a breaking change, as `CommunityDescription` will still arrive as a separate message. However, there could be timing issues. It's better to be safe than sorry, so I'll kindly ask the QAs to test requests to join between the old and new versions for any regressions.